presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1761,6 +1778,31 @@ private JoinFilterFunctionFactory compileJoinFilterFunction(
             return joinFilterFunctionCompiler.compileJoinFilterFunction(translatedFilter, buildLayout.size());
         }
 
+        private Supplier<PageFilter> compileNestedLoopJoinPageFilterSupplier(
```

Comment:
why not reuse `LocalExecutionPlanner.Visitor#compileJoinFilterFunction`?

Reply:
There a couple of reasons why I couldn't use compileJoinFilterFunction as is:

(1) `compileJoinFilterFunction` returns hash join-specific JoinFilterFunction(Factory) which works with "addresses": `boolean filter(int leftAddress, int rightPosition, Page rightPage);`

(2) hash join places build columns before probe columns while nested loop join does the opposite

However, I'm seeing that the new method and compileJoinFilterFunction have some common code. I'll extract it into a helper method. Let me know if that works better.

Reply:
> However, I'm seeing that the new method and compileJoinFilterFunction have some common code. I'll extract it into a helper method. Let me know if that works better.

I wouldn't assume this will improve readability, but sure, give it a try!